### DISCUSSION TITLE
Odd behavior when dropping and then subsequently creating a collection

### DISCUSSION THREAD

**Discussion by MOVzeroOne** [milvus-io/milvus#35603] [Q&A and General discussion]
I use milvusdb/milvus:v2.4.5 docker in docker with  (pymilvus.__version__ == 2.4.5)

```
from pymilvus import MilvusClient

client = MilvusClient(
    uri="http://localhost:19530"
)
client.drop_collection(collection_name="demo")

client.create_collection(collection_name="demo",dimension=3,auto_id=True)

client.insert("demo",[{"vector":[1,2,3],"data":"hello world"}])

print(client.search("demo",[[1,2,3]],output_fields=["data"]))
```
results in ```data: ['[]']``` 

where 

```
from pymilvus import MilvusClient

client = MilvusClient(
    uri="http://localhost:19530"
)

client.create_collection(collection_name="demo",dimension=3,auto_id=True)

client.insert("demo",[{"vector":[1,2,3],"data":"hello world"}])

print(client.search("demo",[[1,2,3]],output_fields=["data"]))
```
results in ```data: ["[{'id': 451978317871448296, 'distance': 0.9999999403953552, 'entity': {'data': 'hello world'}}]"] ```

(I get also the same behavior if I first check if it already  exists and then drop it, as is shown in the quick-start tutorial) 
This feels like it isn't supposed to work like this, what is the proper way to drop a collection and recreating it. Is this intended behavior or might it be a bug?

**Comment by yhmo** [comment]
When you call collection.insert(), the data is received by milvus proxy node, and passed to a message queue(in cluster, Pulsar/Kafka is the message queue; in standalone, message queue is an embedded RocksDB). Once the data is stored into the message queue, the insert() API returns. The data is asynchronously consumed by querynode/datanode. This behavior is like a WAL(write-ahead-log) machinery. Only after the data is consumed by querynode, it is visible to search/query requests.

So, after the insert() is returned, it doesn't mean the data is visible to search/query requests. Data visibility is controlled by consistency level: https://milvus.io/docs/consistency.md#Consistency-levels

When you create a collection, the default consistency level is "Bounded", which means data within a few seconds time window may be invisible.
To ensure all the inserted data is visible, set the consistency level to be "Strong":
```
results = client.search(collection_name="demo",
                            data=[[1,2,3]],
                            anns_field="vector",
                            consistency_level="Strong",
                            output_fields=["data"],
                            )
```
